environment/src/main/java/com/sequenceiq/environment/events/EnvironmentStructuredFlowEventFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.environment.events;
+
+import static com.sequenceiq.cloudbreak.common.exception.NotFoundException.notFound;
+import static com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType.FLOW;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.structuredevent.event.FlowDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPOperationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredFlowEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredNotificationEvent;
+import com.sequenceiq.cloudbreak.structuredevent.service.CDPStructuredFlowEventFactory;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.flow.ha.NodeConfig;
+
+@Component
+public class EnvironmentStructuredFlowEventFactory implements CDPStructuredFlowEventFactory {
+
+    @Inject
+    private Clock clock;
+
+    @Inject
+    private EnvironmentService environmentService;
+
+    @Inject
+    private NodeConfig nodeConfig;
+
+    @Value("${info.app.version:}")
+    private String serviceVersion;
+
+    @Override
+    public CDPStructuredFlowEvent createStructuredFlowEvent(Long resourceId, FlowDetails flowDetails, Boolean detailed) {
+        Optional<Environment> environmentOptional = environmentService.findEnvironmentById(resourceId);
+        Environment environment = environmentOptional.orElseThrow(notFound("Environment", resourceId));
```

Comment:
I would prefer if you would add an `getEnvironmentById` method to the `EnvironmentService` and that could throw an exception as I see this kind of usage a lot in the code generally